### PR TITLE
Add support to select preset in credentials step of KubeOne wizard

### LINKED ISSUE #5502: Add support for presets in import KubeOne cluster wizard
Add support for presets in import KubeOne cluster wizard. A new dropdown should be added on Credentials step which displays list of presets for selected provider. Presets can be loaded from `/api/v2/providers/{provider_name}/presets?disabled=false`.

### PR DESCRIPTION
**Pull request by Waseem826**
**What this PR does / why we need it**:
Add support to select preset in credentials step of KubeOne wizard.

**AWS:**

screenshot-localhost_8000-2023 01 04-16_36_31

**GCP:**

screenshot-localhost_8000-2023 01 04-16_37_27

**Azure:**

screenshot-localhost_8000-2023 01 04-16_38_02

**Summary Step:**

screenshot-localhost_8000-2023 01 04-16_36_54

**Which issue(s) this PR fixes**:

Fixes #5502 

**What type of PR is this?**

/kind feature

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Add support to select preset in credentials step of KubeOne wizard.
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Comment by Waseem826** [comment]
/hold
Wait for backend changes

File: modules/web/src/app/kubeone-wizard/steps/credentials/provider/basic/aws/component.ts [hunk 1]
```diff
@@ -72,12 +77,32 @@ export class KubeOneAWSCredentialsBasicComponent extends BaseFormValidator imple
...
+      .subscribe(preset => Object.values(Controls).forEach(control => this._enable(!preset, control)));
+
     merge(this.form.get(Controls.AccessKeyID).valueChanges, this.form.get(Controls.AccessKeySecret).valueChanges)
       .pipe(distinctUntilChanged())
       .pipe(takeUntil(this._unsubscribe))
       .subscribe(_ => (this._clusterSpecService.cluster = this._getClusterEntity()));
   }
 
+  private _enable(enable: boolean, name: string): void {
+    if (enable && this.form.get(name).disabled) {
```
Comment (ahmadhamzh) [inline modules/web/src/app/kubeone-wizard/steps/credentials/provider/basic/aws/component.ts:97]: `string` to `Controls`

File: modules/web/src/app/kubeone-wizard/steps/credentials/provider/basic/aws/component.ts [hunk 2]
```diff
@@ -72,12 +77,32 @@ export class KubeOneAWSCredentialsBasicComponent extends BaseFormValidator imple
...
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(preset => Object.values(Controls).forEach(control => this._enable(!preset, control)));
+
     merge(this.form.get(Controls.AccessKeyID).valueChanges, this.form.get(Controls.AccessKeySecret).valueChanges)
       .pipe(distinctUntilChanged())
       .pipe(takeUntil(this._unsubscribe))
       .subscribe(_ => (this._clusterSpecService.cluster = this._getClusterEntity()));
   }
 
+  private _enable(enable: boolean, name: string): void {
```
Comment (ahmadhamzh) [inline modules/web/src/app/kubeone-wizard/steps/credentials/provider/basic/aws/component.ts:96]:
```suggestion
  private _enable(enable: boolean, name: Controls): void {
```

Comment (ahmadhamzh) [reply modules/web/src/app/kubeone-wizard/steps/credentials/provider/basic/aws/component.ts:96]: so this method been used a lot in every where isn't better to use a generic method, or it's too late for that ?

File: modules/web/src/app/core/services/kubeone-wizard/kubeone-presets.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    this.presetChanges.next(preset);
+  }
+
+  enablePresets(enable: boolean): void {
+    this.presetStatusChanges.next(enable);
+  }
+
+  presets(provider: NodeProvider): Observable<PresetList> {
+    const url = `${environment.newRestRoot}/providers/${provider}/presets?disabled=false`;
+    return this._http.get<PresetList>(url);
```
Comment (ahmadhamzh) [inline modules/web/src/app/core/services/kubeone-wizard/kubeone-presets.ts:47]: do we need to add catch error ?

Comment (Waseem826) [reply modules/web/src/app/core/services/kubeone-wizard/kubeone-presets.ts:47]: No, I don't think so. We don't have it in other places either.

File: modules/web/src/app/kubeone-wizard/steps/credentials/provider/basic/aws/component.ts (hunk 2 above)
Comment (Waseem826) [reply modules/web/src/app/kubeone-wizard/steps/credentials/provider/basic/aws/component.ts:96]: Yea, at some point we'll need to move this to some shared place.

**Pull request merged**